open-craft/opencraft
# [BB-3686] Change ANSIBLE_APPSERVER_REQUIREMENTS_PATH to requirements_ansible_2.8.17.txt

Changes `ANSIBLE_APPSERVER_REQUIREMENTS_PATH` to `requirements_ansible_2.8.17.txt` and adds migration to update default value in the database.

Currently, when we want to run koa/master Open edX app server, we have to use a separate branch from `ansible-playbooks` repo. So, we always need to rebase this branch from master. This PR solves this issue and enables Ansible 2.8.17 for `appserver.yml` playbook by default.

With fixes that are already merged in `master`, Ansible 2.8.17 has proven to be stable for Ubuntu 16.04 and 20.04, so this change is safe.

**JIRA tickets**:
- [BB-3686](https://tasks.opencraft.com/browse/BB-3686)

**Dependencies**:
- Depends on https://github.com/open-craft/ansible-playbooks/pull/202.

**Testing instructions**:

1. Attach to the stage Ocim screen, and switch window to `bash`.
2. Checkout branch to `0x29a/bb3686/change_default_appserver_requirements_file`.
3. Switch window to `ipython` and restart the shell.
4. Ensure that `settings.ANSIBLE_APPSERVER_REQUIREMENTS_PATH` equals to `requirements_ansible_2.8.17.txt`.
5. Switch window to `bash` and run `make migrate`.
6. After migration is finished, switch back to `ipython` window and create test instance: `production_instance_factory(sub_domain='bb3686')`.
7. Go to https://stage.manage.opencraft.com/admin/, open settings for the instance with `bb3686` sub-domain.
8. **You should see that** `Ansible appserver requirements path` **is equal to** `requirements_ansible_2.8.17.txt`.
9. Change `Ansible appserver version` to `0x29a/bb3686/separe_requirements_file_for_ansible2.8.17`.
10. Leave `Openstack server base image` as is. Our goal is to check that everything works for current default release, `juniper.3`.
11. Create app server for this instance and ensure that it provisioned fine.


**Reviewers**
